### PR TITLE
Documentation: coding style setting with examples

### PR DESCRIPTION
**Pull request by paryoja**
### What is the context for this pull request?

 - **Tracking Issue**: #426 

### What changes were proposed in this pull request?

Add detailed setting for scalafmt with Intellij.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Not required.

### CODE REVIEW

**Comment by paryoja** [comment]
Added guideline / sample image to reference.
I'm not sure is it sufficient.

And I added some image that might help me at the first time I started to setup my Intellij project. 

@imback82 do you think it needs more details? I think Project Structure is a bit too much since it will be displayed at the front page of the github project.

**Review by sezruby — COMMENTED** [review]
I think it might be enough to add a link for the image for reference, at the front page.

For reviewers, links for md rendering:
https://github.com/paryoja/hyperspace/blob/document/coding_style/README.md
https://github.com/paryoja/hyperspace/blob/document/coding_style/docs/coding-guidelines/scala-coding-style.md

File: docs/coding-guidelines/scala-coding-style.md [hunk 1]
```diff
@@ -8,6 +8,13 @@ Scala Coding Style
 > Note: Databricks Style Guide will be preferred for conflict resolutions.
 
 * For formatting, [scalafmt](https://scalameta.org/scalafmt) is used with the custom configuration (found in [/dev/.scalafmt.conf](/dev/.scalafmt.conf))
-  * Installation of `scalafmt` can be found [here](https://scalameta.org/scalafmt/docs/installation.html)
+  * Set scalafmt as the formatter
+    * Open Preferences (Settings if you use Windows) > Editor > Code Style > Scala
+    * Select Scalafmt as the formatter
+    * Check 'Reformat on file save'
+    * Select ./dev/.scalafmt.conf for Configuration
```
Comment (sezruby) [inline docs/coding-guidelines/scala-coding-style.md:15]: nit
```suggestion
    * Select `./dev/.scalafmt.conf` for Configuration
```

File: README.md [hunk 2]
```diff
@@ -28,35 +28,36 @@ This repository contains symbolic links which don't work properly on Windows. To
...
 
 You can use the built-in sbt shell in IntelliJ without any problems. However, the built-in "Build Project" command may not work. To fix the issue, go to Project Structure -> Project Settings -> Modules and follow these steps:
 
-- Mark `src/main/scala` and `src/main/scala-spark2` as "Sources" and `src/test/scala` and `src/test/scala-spark2` as "Tests" for the `spark2_4` module
-- Mark `src/main/scala` and `src/main/scala-spark3` as "Sources" and `src/test/scala` and `src/test/scala-spark3` as "Tests" for the `spark3_0` module
-- Remove the `root` and `hyperspace-sources` modules
+- Mark `src/main/scala` and `src/main/scala-spark2` as "Sources" and `src/test/scala` and `src/test/scala-spark2` as "Tests" for the `spark2_4` module.
+- Mark `src/main/scala` and `src/main/scala-spark3` as "Sources" and `src/test/scala` and `src/test/scala-spark3` as "Tests" for the `spark3_0` module.
+- Remove the `root` and `hyperspace-sources` modules.
+- An example of [Project Structure.](./docs/assets/images/project_structure.png)
```
Comment (sezruby) [inline README.md:41]: nit
```suggestion
- An example of [Project Structure](./docs/assets/images/project_structure.png)
```

**Review by sezruby — APPROVED** [review]
LGTM thanks @paryoja! 
@clee704 @imback82 please leave any comment on this PR if you have any.

**Pull request merged**